src_c/rect.c

```diff
@@ -421,6 +421,38 @@ pg_rect_inflate_ip(pgRectObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+pg_rect_scale_by_ip(pgRectObject *self, PyObject *args)
+{
+    float factor_x, factor_y = 0;
+
+    if (!PyArg_ParseTuple(args, "f|f", &factor_x, &factor_y)) {
+        return NULL;
+    }
+
+    factor_x = factor_x < 0 ? -factor_x : factor_x;
+    factor_y = factor_y < 0 ? -factor_y : factor_y;
+
+    factor_y = (factor_y > 0) ? factor_y : factor_x;
+
+    self->r.x =
+        (int)(self->r.x + (self->r.w / 2) - (self->r.w * factor_x / 2));
+    self->r.y =
+        (int)(self->r.y + (self->r.h / 2) - (self->r.h * factor_y / 2));
+    self->r.w = (int)(self->r.w * factor_x);
+    self->r.h = (int)(self->r.h * factor_y);
+    Py_RETURN_NONE;
+}
+
+static PyObject *
+pg_rect_scale_by(pgRectObject *self, PyObject *args)
+{
+    pgRectObject *returnRect = (pgRectObject *)_pg_rect_subtype_new4(
```

Comment:
`returnRect` <- is there a naming convention for variables? 
It looks like all lowercase and without underline thought.

I know, I'm inconsistent right now, line 427, should probably be `float, factorx, factory = 0;` and `returnrect` here.